### PR TITLE
[GEMM][Tutorial] Refine test_correctness

### PR DESCRIPTION
**Pull request by zhanglx13**
- Check correctness of what is benchmarked
- Add capability to check col_a and col_b
  - But only check col_a=False, col_b=True for now
- Only benchmark col_a=False, col_b=True for now
- Remove in='int8', out='int8' due to too large error

Leftover from https://github.com/ROCmSoftwarePlatform/triton/pull/456

### CODE REVIEW

**Review by jayfurmanek — APPROVED** [review]
LGTM

File: python/perf-kernels/03-matrix-multiplication-all-types.py [hunk 1]
```diff
@@ -205,37 +208,50 @@ def copy_kernel(input_ptr, output_ptr, n_elements, BLOCK_SIZE: tl.constexpr):
...
+        torch_output = torch.matmul(a_fp16, b_fp16)
+    else:
+        torch_output = torch.matmul(a, b)
+    #print(f"triton_output={c}")
+    #print(f"torch_output={torch_output}")
     rtol = 0 if torch.version.hip is None else 1e-2
-    if torch.allclose(c.to(torch.float16), torch_output, atol=5e-2, rtol=rtol):
-        print("✅ Triton and Torch match")
+    if in_dtype == 'int8':
+        torch.testing.assert_close(c.to(torch.float16), torch_output, atol=5e-2, rtol=rtol)
```
Comment (scxiao) [inline python/perf-kernels/03-matrix-multiplication-all-types.py:252]: If we use a smaller K and do not use splitK, maybe we can use smaller tolerances. 
For `int8` data type, can we cast to int32 type for comparison, so we can use 0 as tolerance.

Comment (zhanglx13) [reply python/perf-kernels/03-matrix-multiplication-all-types.py:252]: > If we use a smaller K and do not use splitK, maybe we can use smaller tolerances.

I assume you mean smaller BLOCK_K. 
We cannot pick the tiling and parameter here. They are picked by the autotuner. We want to check the correctness of the configs that is used in the benchmark.

> For int8 data type, can we cast to int32 type for comparison, so we can use 0 as tolerance

Let me give it a shot

Comment (zhanglx13) [reply python/perf-kernels/03-matrix-multiplication-all-types.py:252]: > For int8 data type, can we cast to int32 type for comparison, so we can use 0 as tolerance

It doesn't help. 
For int8 input and int8 output, it seems the precision is lost a lot.

**Pull request merged**